src/librustc_incremental/persist/load.rs

```diff
@@ -103,41 +131,50 @@ fn delete_dirty_work_product(tcx: TyCtxt,
     work_product::delete_workproduct_files(tcx.sess, &swp.work_product);
 }
 
-pub fn load_dep_graph(sess: &Session) -> PreviousDepGraph {
-    let empty = PreviousDepGraph::new(SerializedDepGraph::new());
-
-    if sess.opts.incremental.is_none() {
-        return empty
-    }
-
-    if let Some((bytes, start_pos)) = load_data(sess, &dep_graph_path(sess)) {
-        let mut decoder = Decoder::new(&bytes, start_pos);
-        let prev_commandline_args_hash = u64::decode(&mut decoder)
-            .expect("Error reading commandline arg hash from cached dep-graph");
-
-        if prev_commandline_args_hash != sess.opts.dep_tracking_hash() {
-            if sess.opts.debugging_opts.incremental_info {
-                println!("[incremental] completely ignoring cache because of \
-                          differing commandline arguments");
+/// Launch a thread and load the dependency graph in the background.
+pub fn load_dep_graph(sess: &Session, time_passes: bool) ->
+    std::thread::JoinHandle<LoadResult<PreviousDepGraph>>
+{
+    let is_incremental = sess.opts.incremental.is_some();
+    let report_incremental_info = sess.opts.debugging_opts.incremental_info;
+    let path = dep_graph_path_from(&sess.incr_comp_session_dir());
```

Comment:
This line might be crashing if the dependency graph is enabled but we don't have a caching directory. This can happen in the failing test cases. I'll verify this later.